examples/multipole.py

```diff
@@ -0,0 +1,145 @@
+# Example illustrating the novel boundary sweeping functions (extrude/revolve) introduced in v0.8.0
+# at the hand of a simple simulation of a multipole
+
+import numpy as np
+
+import traceon.geometry as G
+
+import traceon.solver as S
+import traceon.excitation as E 
+import traceon.plotting as P
+import traceon.tracing as T
+
+
+# multipole configuration
+ORDER = 2 # A second-order multipole is a quadrupole
+THICKNESS = 0.1
+IN_RADIUS = 0.3
+OUT_RADIUS = 0.5
+SPACING = 0.05
+
+# boundary configuration
+B_RADIUS = 1.
+B_HEIGHT = 2.
+
+
+# Before v0.8.0
+# 1. Initialize a path
+# 2. Revolve it
+# 3. Add side pieces
+
+multipole_pieces = []
+
+for n in range(1, 2 * ORDER + 1):
+    angle = 2 * np.pi / (2 * ORDER)
+    piece = (
+        G.Path.rectangle_xz(IN_RADIUS, OUT_RADIUS, -THICKNESS/2, THICKNESS/2) 
+        .rotate(Rz=n * angle)
+        .revolve_z(angle)
+        .move(
+            dx=SPACING * np.cos((n + 0.5) * angle),
+            dy=SPACING * np.sin((n + 0.5) * angle),
+        )
+    )
```

Comment:
Please don't put `)` on their own line, as that is not the style in the codebase. Same for the other code below. You can simply do:
```Python
        .move(
            dx=SPACING * np.cos((n + 0.5) * angle),
            dy=SPACING * np.sin((n + 0.5) * angle)))
```